### PR TITLE
Add new testing infra

### PR DESCRIPTION
**Pull request by kianenigma**
- adds the staking miner node 
- some prototyping around tests, not finished yet.

### CODE REVIEW

**Comment by kianenigma** [comment]
We better get this merged or closed soon, been around for a long long time.

**Comment by niklasad1** [comment]
yeah, I'll give it a go tomorrow but the node template needs to be updated because of the weights v2 stuff ^^

File: staking-miner-test-node/runtime/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,885 @@
...
+	pub fn set(len: u64) {
+		frame_support::storage::unhashed::put(BLOCK_WEIGHT_KEY, &len);
+	}
+
+	pub struct ConfigurableBlockWeight;
+	impl Get<limits::BlockWeights> for ConfigurableBlockWeight {
+		fn get() -> limits::BlockWeights {
+			// limits::BlockWeights::simple_max(frame_support::weights::Weight::from_ref_time(get()))
+			//
+			// TODO(niklasad1): I don't understand why this panics, seems to works in substrate
```
Comment (niklasad1) [inline staking-miner-test-node/runtime/src/lib.rs:184]:
```suggestion
			// TODO(niklasad1): I don't understand why this panics, seems to work in substrate
```

**Comment by kianenigma** [comment]
I would love to see this closed or merged or generally out of my PR list for now @niklasad1 :D

File: staking-miner-test-node/runtime/src/pallet_config_block.rs [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+use frame_support::{
+	dispatch::DispatchResult,
+	weights::{constants::WEIGHT_PER_SECOND, Weight},
+};
+use sp_std::marker::PhantomData;
+
+// Re-export pallet items so that they can be accessed from the crate namespace.
+pub use pallet::*;
+
+// Definition of the pallet logic, to be aggregated at runtime definition through
```
Comment (kianenigma) [inline staking-miner-test-node/runtime/src/pallet_config_block.rs:15]: some of these docs can go away.

**Review by niklasad1 — APPROVED** [review]

**Comment by niklasad1** [comment]
NOTE, this is not enabled in the CI yet but I will do a follow to try implement some "real trimming" tests and enable it in the CI.

Just a relief to merge it :)

**Pull request merged**